gnomad/assessment/validity_checks.py

```diff
@@ -857,12 +863,53 @@ def vcf_field_check(
     return True
 
 
+def compare_related_global_and_row_lengths(
+    t: Union[hl.MatrixTable, hl.Table],
+    len_comp_globals_rows: Dict[str, List[str]] = None,
+) -> None:
+    """
+    Compare the lengths of global and row annotations.
+
+    :param t: Input MatrixTable or Table.
+    :param len_comp_globals_rows: Dictionary with row annotation (key) and list of associated global annotations (value) to compare.
+    :return: None
+    """
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
+    for row_field, global_fields in len_comp_globals_rows.items():
+        row_len = len(t.head(1)[row_field].collect()[0])
```

Comment:
I don't think this will make a difference in performance, but you could do this above
```
t = t.rows() if isinstance(t, hl.MatrixTable) else t
t.head(1)
```
since you only ever need to access the first row in the for loop below

Reply:
Yeah I like that better